public/models/Player.js

```diff
@@ -12,39 +12,123 @@ class Player extends GameObject { // Marianna
+          if (x % map.tileWidth) {
+            collision = this.handleCollision(map.tiles[row - 1][column + 1], row - 1, column + 1, onlyWallCollision) || collision;
+            // console.log('check row', row - 1, 'column', column + 1);
+          }
+        }
+        break;
+      case 'right':
+        if (x % map.tileWidth) {
+          collision = this.handleCollision(map.tiles[row - 1][column + 1], row - 1, column + 1, onlyWallCollision);
+          // console.log('check row', row - 1, 'column', column + 1);
+          if (y % map.tileHeight) {
+            collision = this.handleCollision(map.tiles[row][column + 1], row, column + 1, onlyWallCollision) || collision;
+            // console.log('check row', row, 'column', column + 1);
+          }
+        } else {
+          collision = this.handleCollision(map.tiles[row - 1][column], row - 1, column, onlyWallCollision);
+          // console.log('check row', row - 1, 'column', column);
+          if (y % map.tileHeight) {
+            collision = this.handleCollision(map.tiles[row][column], row, column, onlyWallCollision) || collision;
+            // console.log('check row', row, 'column', column + 1);
+          }
+        }
+        break;
+      case 'left':
+        collision = this.handleCollision(map.tiles[row - 1][column], row - 1, column, onlyWallCollision);
+        // console.log('check row', row - 1, 'column', column);
+        if (y % map.tileHeight) {
+          collision = this.handleCollision(map.tiles[row][column], row, column, onlyWallCollision) || collision;
+          // console.log('check row', row, 'column', column);
+        }
+        break;
+      default:
     }
+    return collision;
+  }
+
+  handleCollision(block, row, column, onlyWallCollision) {
+    // check only wall collisions
+    if (block !== 1 && onlyWallCollision) block = '';
+    block = String(block);
+    switch (block) {
+      case '1':
+        // console.log("wall", row, column);
+        return true;
+      case '2':
+        // console.log("goal", row, column);
+        this.playerIsDone();
+        break;
+      case (block.match(/^4/)?.input):
+        // console.log("coin", row, column);
+        this.handleCoinCollision(block, row, column);
+        break;
+      default:
+    }
+  }
+
+  // Marianna
+  // set score when player finished the game
+  // show score of all players and move to spectator mode
+  playerIsDone() {
+    this.isDone = true;
+    // maximum time - elapsed time
+    let timeScore = map.timeLimit - (new Date().getTime() - startTime);
+    // if final number is negative, set it to 0;
+    timeScore = timeScore > 0 ? timeScore : 0;
+    this.score += timeScore;
+    endScreen.setAttribute('style', 'display:block');
+    socket.emit('playerFinished', this);
+    this.draw = () => {};
+    // playing false so movement keys get disabled
+    playing = false;
+    window.addEventListener('keyup', changeSpectator);
+  }
+
+  handleCoinCollision(block, row, column) {
```

Comment:
what is block here?

Reply:
like block type with which the player is colliding such as wall or goal or coin. Should I rename it maybe to tile? hmm

Reply:
no no I just wasn't sure